enumerable_methods.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # my_each
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  # my_select
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |a|
+      arr.push(a) if yield(a)
+    end
+    arr
+  end
+
+  # my_all
+  def my_all?(arg = nil) # rubocop:disable Layout/TrailingWhitespace
+    if block_given?
+      my_each { |a| return false unless yield(a) }
+      return true
+    end
+    if arg.class == Regexp || arg.class == String
+      my_each { |a| return false unless a.match(arg) }
+      return true
+    elsif arg.class == Class
+      my_each { |a| return false unless a.class == arg || a.class.superclass == arg }
+      return true
+    elsif arg.nil? && !block_given? && !empty?
+      my_each { |a| return false if a.nil? || a == false }
+    elsif (arg.class != Class || arg.class != Regexp) && !arg.nil?
+      my_each { |a| return true unless a != arg }
+    end
+    true
+  end
+
+  # my_any
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |a| return true if yield(a) }
+      return false
+    end
+    if arg.class == Regexp || arg.class == String
+      my_each { |a| return true if a.match(arg) }
+      return false
+    elsif arg.class == Class
+      my_each { |a| return true if a.class == arg || a.class.superclass == arg }
+      return false
+    elsif arg.nil? && !block_given? && !empty?
+      my_each { |a| return true unless a.nil? || a == false }
+    end
+    false
+  end
+
+  # my_none
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |a| return false if yield(a) }
+      return true
+    end
+    if arg.class == Regexp || arg.class == String
+      my_each { |a| return false if a.match(arg) }
+      return true
+    elsif arg.class == Class
+      my_each { |a| return false if a.class == arg || a.class.superclass == arg }
+      return true
+    elsif arg.nil? && !block_given? && !empty?
+      return false
+    elsif !arg.nil?
+      my_each { |a| return false if a == arg }
+    end
+    true
+  end
```

Comment:
enumerables my_none? when no block or argument is given returns true only if none of the collection members is true
     Failure/Error: expect(false_array.my_none?).to be true
     
       expected true
            got false